src/main/java/com/minecolonies/coremod/commands/colonycommands/MakeNotAutoDeletableCommand.java

```diff
@@ -50,36 +64,47 @@ public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final En
     }
 
     @Override
-    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
     {
-        if (sender instanceof EntityPlayer && !isPlayerOpped(sender))
-        {
-            sender.sendMessage(new TextComponentString("Must be OP to use command"));
-            return;
-        }
-        else if (sender instanceof TileEntity)
-        {
-            return;
-        }
+        final Colony colony = actionMenu.getColonyForArgument("colony");
+        final boolean canBeDeleted = actionMenu.getBooleanValueForArgument("canBeDeleted", false);
+        executeShared(server, sender, colony, canBeDeleted);
+    }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
         if (args.length < NUMBER_OR_ARGS_REQUIRED)
         {
             sender.sendMessage(new TextComponentString(NOT_ENOUGH_ARGUMENTS));
             return;
         }
 
-        sender.sendMessage(new TextComponentString(Arrays.toString(args)));
-
         final int colonyId;
         colonyId = Integer.parseInt(args[0]);
         final Colony colony = ColonyManager.getColony(colonyId);
 
+        if (colony == null)
+        {
+            sender.sendMessage(new TextComponentString(String.format(NO_COLONY_FOUND_MESSAGE_ID, colonyId)));
+            return;
+        }
+
         final boolean canBeDeleted;
         canBeDeleted = Boolean.parseBoolean(args[1]);
 
-        if (colony == null)
+        executeShared(server, sender, colony, canBeDeleted);
+    }
+
+    private void executeShared(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final Colony colony, final boolean canBeDeleted)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "server". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)